src/Compatibility/Microsoft.DotNet.ApiCompatibility/Rules/RuleDriver.cs

```diff
@@ -0,0 +1,92 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.s
+
+using Microsoft.DotNet.ApiCompatibility.Abstractions;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.DotNet.ApiCompatibility.Rules
+{
+    public interface IRuleDriverFactory
+    {
+        IRuleDriver GetRuleDriver();
+    }
+
+    public class RuleDriverFactory : IRuleDriverFactory
+    {
+        private RuleDriver _driver;
+        public IRuleDriver GetRuleDriver()
+        {
+            if (_driver == null)
+                _driver = new RuleDriver();
+
+            return _driver;
+        }
+    }
+
+    public interface IRuleDriver
```

Comment:
Naming: driver here was confusing to me.  Consider `runner`, `controlller`, `exector`.